src/EnergyPlus/HeatBalFiniteDiffManager.cc

```diff
@@ -1573,14 +1573,21 @@ namespace HeatBalFiniteDiffManager {
             auto TDT_i(TDT(i));
             auto const TDT_p(TDT(i + 1));
 
+            Real64 Tgndsurface = 0.0;
+            if (state.dataSurface->UseSurfPropertyGndSurfTemp(Surf)) {
+                Tgndsurface = state.dataSurface->GroundSurfsProperty(Surf).SurfsTempAvg;
+            } else {
+                Tgndsurface = state.dataMstBal->TempOutsideAirFD(Surf);
+            }
+
```

Comment:
If multiple ground surfaces property object is specified, then sets ground surfaces object ground surface temperature as boundary condition instead of the default.